config/software/datadog-agent.rb

```diff
@@ -25,7 +31,18 @@
   mkdir  "#{install_dir}/agent/"
 
   # Agent code
+  mkdir "#{install_dir}/agent/checks.d"
   copy 'checks.d', "#{install_dir}/agent/"
+  checks = Dir["#{integrations_dir}/*"]
```

Comment:
This will pick up files as well, not only directories. `Dir.glob("#{integrations_dir}/*/"` should give you what you expect. (I prefer `Dir.glob` over `Dir[]` because it's easier to understand when you don't know Ruby)